### PR TITLE
Add support for pagination at controller level. 

### PR DESCRIPTION
**Pull request by DavideSegullo**
### Changelist
This pr is a continuation of PR https://github.com/dydxprotocol/v4-chain/pull/1244.
In particular, it modifies the controllers of fills, trades, and transfers to support paging in REST services.

### Test Plan
Tests are delivered with the features.
We have not tested the environment with a full live (development) environment, but have performed unit tests.
We encourage you to perform full (integration) testing before merging.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Introduced pagination parameters (`page`, `pageSize`, `totalResults`, `offset`) to API endpoints for fills, trades, and transfers.
  - Enhanced API responses to include pagination details.

- **Bug Fixes**
  - Ensured trades and fills are fetched in descending order by `createdAtHeight` and correctly paginated.

- **Documentation**
  - Updated API documentation to reflect new pagination parameters and response fields.

- **Tests**
  - Added comprehensive test cases to verify pagination functionality for fills, trades, and transfers.

### CODE REVIEW

**Review by Christopher-Li — APPROVED** [review]

File: indexer/services/comlink/src/controllers/api/v4/fills-controller.ts [hunk 1]
```diff
@@ -77,6 +88,7 @@ class FillsController extends Controller {
           ? createdBeforeOrAtHeight.toString()
           : undefined,
         createdBeforeOrAt,
+        page,
       },
       [QueryableField.LIMIT],
```
Comment (dydxwill) [inline indexer/services/comlink/src/controllers/api/v4/fills-controller.ts:93]: We need to specify a default ordering. Can you add
`{
        orderBy: [[FillColumns.eventId, Ordering.ASC]],
      },`?

File: indexer/services/comlink/src/controllers/api/v4/fills-controller.ts [hunk 2]
```diff
@@ -141,6 +162,7 @@ class FillsController extends Controller {
           ? createdBeforeOrAtHeight.toString()
           : undefined,
         createdBeforeOrAt,
+        page,
       },
```
Comment (dydxwill) [inline indexer/services/comlink/src/controllers/api/v4/fills-controller.ts:166]: Same here. Can you add
`{
        orderBy: [[FillColumns.eventId, Ordering.ASC]],
      },`?

File: indexer/services/comlink/src/controllers/api/v4/trades-controller.ts [hunk 3]
```diff
@@ -62,6 +68,7 @@ class TradesController extends Controller {
           ? createdBeforeOrAtHeight.toString()
           : undefined,
         createdBeforeOrAt,
+        page,
       },
       [QueryableField.LIQUIDITY, QueryableField.CLOB_PAIR_ID, QueryableField.LIMIT],
```
Comment (dydxwill) [inline indexer/services/comlink/src/controllers/api/v4/trades-controller.ts:73]: We need to specify a default ordering. Can you add
`{
        orderBy: [[FillColumns.eventId, Ordering.ASC]],
      },`?

File: indexer/services/comlink/src/controllers/api/v4/transfers-controller.ts [hunk 4]
```diff
@@ -77,6 +81,7 @@ class TransfersController extends Controller {
             ? createdBeforeOrAtHeight.toString()
             : undefined,
           createdBeforeOrAt,
+          page,
         },
         [QueryableField.LIMIT],
         {
```
Comment (dydxwill) [inline indexer/services/comlink/src/controllers/api/v4/transfers-controller.ts:87]: Can you change the ordering to be by eventId desc here? To make transfer ordering deterministic if they occurred at the same height.

File: indexer/services/comlink/src/controllers/api/v4/transfers-controller.ts [hunk 5]
```diff
@@ -165,6 +179,7 @@ class TransfersController extends Controller {
...
```
Comment (dydxwill) [inline indexer/services/comlink/src/controllers/api/v4/transfers-controller.ts:185]: Can you change the ordering to be by eventId desc here? To make transfer ordering deterministic if they occurred at the same height.

Comment (DavideSegullo) [reply indexer/services/comlink/src/controllers/api/v4/transfers-controller.ts:190]: Done https://github.com/dydxprotocol/v4-chain/pull/1420/commits/2fd8d09f2b96b3cb0e71f79b02c10d7b8bba992e#diff-02ea2aa83d63cb59bf5894f9412f98d5c06e986cb5827f1a757a421173972474R192

File: indexer/services/comlink/src/controllers/api/v4/transfers-controller.ts (hunk 4 above)
Comment (DavideSegullo) [reply indexer/services/comlink/src/controllers/api/v4/transfers-controller.ts:87]: Done https://github.com/dydxprotocol/v4-chain/pull/1420/commits/2fd8d09f2b96b3cb0e71f79b02c10d7b8bba992e#diff-02ea2aa83d63cb59bf5894f9412f98d5c06e986cb5827f1a757a421173972474R89

File: indexer/services/comlink/src/controllers/api/v4/trades-controller.ts (hunk 3 above)
Comment (DavideSegullo) [reply indexer/services/comlink/src/controllers/api/v4/trades-controller.ts:75]: Done https://github.com/dydxprotocol/v4-chain/pull/1420/commits/2fd8d09f2b96b3cb0e71f79b02c10d7b8bba992e#diff-d8d0c764d2a842bf225a97d2160c105b481d115d5d483347392e0770ff1f404aR76

File: indexer/services/comlink/src/controllers/api/v4/fills-controller.ts (hunk 2 above)
Comment (DavideSegullo) [reply indexer/services/comlink/src/controllers/api/v4/fills-controller.ts:169]: Done https://github.com/dydxprotocol/v4-chain/pull/1420/commits/2fd8d09f2b96b3cb0e71f79b02c10d7b8bba992e#diff-84a6e6fd70f726b166899c1c25ded7304b24411adae6b71622815379318f6d3dR171

File: indexer/services/comlink/src/controllers/api/v4/fills-controller.ts (hunk 1 above)
Comment (DavideSegullo) [reply indexer/services/comlink/src/controllers/api/v4/fills-controller.ts:95]: Done https://github.com/dydxprotocol/v4-chain/pull/1420/commits/2fd8d09f2b96b3cb0e71f79b02c10d7b8bba992e#diff-84a6e6fd70f726b166899c1c25ded7304b24411adae6b71622815379318f6d3dR96

**Review by dydxwill — APPROVED** [review]

**Comment by dydxwill** [comment]
https://github.com/Mergifyio backport release/indexer/v5.x

**Pull request merged**